tests/apps/search/test_query_categories.py

```diff
@@ -0,0 +1,134 @@
+            body=CategoriesIndexer.mapping, doc_type="category", index="test_categories"
+        )
+
+        # Actually insert our categories in the index
+        actions = [
+            {
+                "_id": category["id"],
+                "_index": "test_categories",
+                "_op_type": "create",
+                "_type": "category",
+                "absolute_url": {"en": "en/url"},
+                "description": {"en": "en/description"},
+                "icon": {"en": "en/icon"},
+                "is_meta": False,
+                "logo": {"en": "en/logo"},
+                "nb_children": 0,
+                "path": category["id"],
+                **category,
+            }
+            for category in CATEGORIES
+        ]
+        bulk(actions=actions, chunk_size=500, client=ES_CLIENT)
+        indices_client.refresh()
+
+        response = self.client.get(f"/api/v1.0/{kind:s}/?{querystring:s}")
+        self.assertEqual(response.status_code, 200)
+
+        return json.loads(response.content)
+
+    def test_query_all_categories_by_kind(self, *_):
+        """
+        Make sure all categories of a given kind are returned when there is no text query.
+        """
+        content = self.execute_query(kind="subjects")
+        self.assertEqual(
+            content,
+            {
+                "meta": {"count": 2, "offset": 0, "total_count": 2},
+                "objects": [
+                    {
+                        "icon": "en/icon",
+                        "id": "8399",
+                        "is_meta": False,
+                        "logo": "en/logo",
+                        "nb_children": 0,
+                        "path": "8399",
+                        "title": "Science for beginners",
+                    },
+                    {
+                        "icon": "en/icon",
+                        "id": "8312",
+                        "is_meta": False,
+                        "logo": "en/logo",
+                        "nb_children": 0,
+                        "path": "8312",
+                        "title": "Literature",
+                    },
+                ],
+            },
+        )
+
+    def test_query_categories_by_text_and_kind(self, *_):
+        """
+        Make sure only categories matching the text query and kind are returned.
+        """
+        content = self.execute_query(kind="levels", querystring="query=beginner")
+        self.assertEqual(
+            content,
+            {
+                "meta": {"count": 1, "offset": 0, "total_count": 1},
+                "objects": [
+                    {
+                        "icon": "en/icon",
+                        "id": "8421",
+                        "is_meta": False,
+                        "logo": "en/logo",
+                        "nb_children": 0,
+                        "path": "8421",
+                        "title": "Beginner",
+                    }
+                ],
+            },
+        )
```

Comment:
Maybe add accents to your content to make the same "diacritic insensitive" tests as what you did on the autocomplete endpoint?

Reply:
You should also test the other way around ie that
```python
content = self.execute_query(kind="subjects", querystring="query=Bégînnêr")
```
returns ID 8399.

Reply:
Done 👍